ui/src/app-bar/app-bar-more-menu.tsx

```diff
@@ -53,31 +52,27 @@ const AppBarMoreMenu: React.FC = () => {
 
     const createRoleSelectionOptions = (): (MenuSelectOption | MenuDividerOption)[] => {
         return availableRoles.length > 1
-            ? [
-                  menuDividerOption(),
-                  ...availableRoles.map(createRoleSelectionOption),
-                  menuDividerOption(),
-              ]
+            ? [menuDivider(), ...availableRoles.map(createRoleSelectionOption), menuDivider()]
             : [];
     };
 
     const moreActions: (MenuSelectOption | MenuDividerOption)[] = [
-        menuSelectOption(
+        menuOption(
             () => {
-                setShowMenu(false);
                 navigate('licenses');
             },
             t('app-bar.licenses'),
             'licenses',
+            true,
```

Comment:
Tämä parametri kuvastaa sitä suljetaanko menu klikkauksen jälkeen vai ei. Mietin että tekisinkö tästä optionalin vai en, siten että default olisi `true` (meillä ei koskaan haluta asettaa tätä `false`:ksi tällä hetkellä.) Toinen vaihtoehto olisi, että tämän parametrin jättäisi kokonaan pois ja kaikki menut aina sulkeutuisivat automaattisesti jonkun itemin klikkaamisen jälkeen, mutta toistaiseksi päädyin tähän. Thoughts?

Reply:
Boolean-liput on aina vähän ikäviä kun niistä ei tiedä yhtään miten ne vaikuttavat (const tai enum olisi jälleen se mielekkäämpi vaihtoehto).

Jos meillä ei kerta ole toistaiseksi yhtään käyttöpaikkaa, jossa menu halutaan pitää auki valinnan jälkeen, niin varmaan ois "sane default" se aina sulkea, mutta komponentin puolesta olisi silti mielekästä mahdollistaa tuo erikoistapauksen käyttö myös (eli pidettäisiin parami, mutta default=CLOSE_MENU_AFTER_SELECT [tms])

Reply:
Mmh, hyvä pointti tuo enumin käyttö. Jotenkin tullut itse totuttua siihen, että tällaiset kaksijakoiset toiminnallisuudet parametrisoidaan aina vaan booleaneilla. Jos tuo autoclose päädytään pitämään, niin voisin muuttaa tuon kaksiarvoiseksi enumiksi/vakioksi (esim. tyyliin `type ClickBehaviour = 'CLOSE_ON_CLICK' | 'NONE';` tms., tai jollekin muulle vastaavalle, paremmalle nimelle)

Reply:
Joo, se on yllättävän tyypillistä.

Muistan vaan itse miten raskasta oli yrittää tulkita koodia jossa oli suurinpiirtein listattu 20 argumenttia funktiokutsuille, joista 10 oli booleaneja mitkä vaikuttivat eri tavalla eri paikoissa -- tällaiset yksittäiset vielä jotenkin menettelee, mutta enumit/tms on kyllä tosi jees näihin ja itse tottunut niitä (hieman kantapään kautta) käyttämään melkolailla aina.

Tuossa esimerkissäsi muuten tuo 'NONE' käytännössä vastaa booleania (eli se ei kerro mitään ekstraa verrattuna falseen)

Reply:
Juu, se `NONE` oli sellainen muutaman sekunnin mietinnän jälkeinen heitto demonstroimaan sitä että tämän voisi tehdä jollain muullakin tavalla kuin booleanilla, enkä itsekään ollut vielä kauhean tyytyväinen tuohon koko esimerkkityypin nimeämiseen. Ideana se, että miettisin tuon kunnolla sitten kun sen oikeasti teen.

Mutta tuo kuvaa kokonaisuudessaan vähän vaikeampaa ongelmaa, joka on se, että miten tällaiset "ei mitään / älä tee mitään" -tyyppiset vakiot kuuluisi nimetä. Enumissa tulisi tyyppi eteen kertomaan, mutta preferoin itse tällaisissa vakioita puhtaiden enumien yli (eihän näitä koskaan enumeroida, joten _enumeraatio_ kuulostaa väärältä by default, vaikka siitä onkin jostain syystä tullut jonkinasteinen konventio muissa kielissä.) Mutta joo, mietin tuon vielä kuntoon kun tuohon uudestaan kosken.

Reply:
Mä tulkitsen enumin siten, että se _enumeroi_ mahdolliset vaihtoehdot.

Ja joo, näitä harvemmin _enumeroidaan_, mutta toki satunnaisesti sekin käytössä (esim. testi joka looppaa kaikkien vaihtoehtojen yli)